src/modules/lua.rs

```diff
@@ -0,0 +1,170 @@
+        Err(error) => {
+            log::warn!("Error in module `lua`:\n{}", error);
+            return None;
+        }
+    });
+
+    Some(module)
+}
+
+fn get_lua_version(lua_binary: &str) -> Option<String> {
+    match utils::exec_cmd(lua_binary, &["-v"]) {
+        Some(output) => {
+            if output.stdout.is_empty() {
+                Some(output.stderr)
+            } else {
+                Some(output.stdout)
+            }
+        }
+        None => None,
+    }
+}
+
+fn format_lua_version(lua_stdout: &str) -> Option<String> {
+    // lua -v output looks like this:
+    // Lua 5.4.0  Copyright (C) 1994-2020 Lua.org, PUC-Rio
+
+    // luajit -v output looks like this:
+    // LuaJIT 2.0.5 -- Copyright (C) 2005-2017 Mike Pall. http://luajit.org/
+    let re = Regex::new(LUA_VERSION_PATERN).ok()?;
+    let captures = re.captures(lua_stdout)?;
+    let version = &captures["version"];
+    Some(format!("v{}", version))
+}
+
+#[cfg(test)]
+mod tests {
+    use super::*;
+    use crate::test::ModuleRenderer;
+    use ansi_term::Color;
+    use std::fs::{self, File};
+    use std::io;
+
+    #[test]
+    fn folder_without_lua_files() -> io::Result<()> {
+        let dir = tempfile::tempdir()?;
+        let actual = ModuleRenderer::new("lua").path(dir.path()).collect();
+        let expected = None;
+        assert_eq!(expected, actual);
+        dir.close()
+    }
+
+    #[test]
+    fn folder_with_lua_file() -> io::Result<()> {
+        let dir = tempfile::tempdir()?;
+        File::create(dir.path().join("main.lua"))?.sync_all()?;
+        let actual = ModuleRenderer::new("lua").path(dir.path()).collect();
+        let expected = Some(format!("via {} ", Color::Blue.bold().paint("🌙 v5.4.0")));
+        assert_eq!(expected, actual);
+        dir.close()
+    }
+
+    #[test]
+    fn folder_with_lua_version() -> io::Result<()> {
+        let dir = tempfile::tempdir()?;
+        File::create(dir.path().join(".lua-version"))?.sync_all()?;
+
+        let actual = ModuleRenderer::new("lua").path(dir.path()).collect();
+        let expected = Some(format!("via {} ", Color::Blue.bold().paint("🌙 v5.4.0")));
+        assert_eq!(expected, actual);
+        dir.close()
+    }
+
+    #[test]
+    fn folder_with_lua_folder() -> io::Result<()> {
+        let dir = tempfile::tempdir()?;
+        let lua_dir = dir.path().join("lua");
+        fs::create_dir_all(&lua_dir)?;
+
+        let actual = ModuleRenderer::new("lua").path(dir.path()).collect();
+        let expected = Some(format!("via {} ", Color::Blue.bold().paint("🌙 v5.4.0")));
+        assert_eq!(expected, actual);
+        dir.close()
+    }
+
+    #[test]
+    fn lua_binary_is_luajit() -> io::Result<()> {
+        let dir = tempfile::tempdir()?;
+        File::create(dir.path().join("main.lua"))?.sync_all()?;
+
+        check_luajit_renders(&dir, None);
+        dir.close()
+    }
+
+    fn check_luajit_renders(dir: &tempfile::TempDir, starship_config: Option<toml::Value>) {
```

Comment:
```suggestion
    #[test]
    fn check_luajit_renders(dir: &tempfile::TempDir, starship_config: Option<toml::Value>) {
```
I think this is missing a `#[test]` annotation.

Reply:
I referred to https://github.com/starship/starship/blob/master/src/modules/python.rs#L358.

I think this fn doesn't require annotation because this is called by other tests which has `#[test]` annotation.

Reply:
Sorry I didn't see that. The `assert_eq!(expected, actual);` tripped me up.

Reply:
So the reason that this was done in a separate function in the python module was because we reused the functions across all the tests so we would test all the different scenarios worked correctly, I am not sure if that is what is being done here. Would it make sense to test against both lua binaries in all the tests here too?

Reply:
I merge this function in `lua_binary_is_luajit` because it only is called by `lua_binary_is_luajit`.